pkg/debug/debug.go

```diff
@@ -0,0 +1,289 @@
+// +build debug
+
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"log"
+	"net/http"
+	"time"
+)
+
+// LogBuildDetails logs the build details when the cspc-operator starts.
+func LogBuildDetails() {
+	klog.Info("This is a DEBUG build and should not be used in production")
+}
+
+// StartInjectionServer is a wrapper that starts a REST server that is used to inject errors in the debug build.
+func StartInjectionServer() {
+	go StartServer()
+}
+
+// StartServer starts the injection server.
+func StartServer() {
+	klog.Info("Starting Error Injection API Server...")
+	RegisterRoutes()
+	server := &http.Server{
+		Addr:         ":8080",
+		ReadTimeout:  60 * time.Second,
+		WriteTimeout: 60 * time.Second,
+	}
+
+	log.Println("Injection Server Listening on 8080 ...")
+	server.ListenAndServe()
+}
+
+// RegisterRoutes registers routes for the injection server.
+func RegisterRoutes() {
+	http.HandleFunc("/", index)
+	http.HandleFunc("/inject", inject)
+}
+
+// index is the handler of '/' route.
+func index(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(200)
+	w.Write([]byte("Error Injection API Server Is Running!"))
+}
+
+// inject is the handler of 'inject' route
+func inject(w http.ResponseWriter, r *http.Request) {
+
+	switch r.Method {
+	case "GET":
+		errorInjectionConfigInBytes, err := json.Marshal(EI)
+		if err != nil {
+			klog.Errorf("Failed to marshal error injection config: %s", err.Error())
+			w.WriteHeader(500)
+			w.Write([]byte(fmt.Sprintf("Failed to marshal error injection config: %s", err.Error())))
+			return
+		}
+		w.WriteHeader(200)
+		w.Write(errorInjectionConfigInBytes)
+
+	case "POST":
+		reqBody, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			fmt.Fprintf(w, "Incorrect data or format in body")
+			w.WriteHeader(500)
+			w.Write([]byte(fmt.Sprintf("Incorrect data or format in body: %s", err.Error())))
+			return
+		}
+		json.Unmarshal(reqBody, EI)
+
+	default:
+		fmt.Fprintf(w, "Method Not Allowed")
+		w.WriteHeader(405)
+		w.Write([]byte(fmt.Sprint("Method Not Allowed")))
+
+	}
+}
+
+// CSPCs
+
+// IsCSPCDeleteCollectionErrorInjected returns true if error is injected for
+// CSPC delete collection.
+func (ei *ErrorInjection) IsCSPCDeleteCollectionErrorInjected() bool {
```

Comment:
Can we move this code to different file.. like `cstor_debug.go`? It will be good to have structured code..
```
inject_client.go
cstore_release.go
cstore_debug.go
inject_server.go
inject_server_release.go
types.go
```

Reply:
Sure. Sounds good. Will do it in the next PR.

Reply:
Made a issue for this in maya to track.